src/charm.py

```diff
@@ -64,6 +69,29 @@ def _on_config_changed(self, event) -> None:
         self.model.unit.status = MaintenanceStatus(msg)
         self.configure_workload_container(event)
 
+    def _report_visits_by_ip_action(self, event: ActionEvent) -> None:
+        """Handle the report-visits-by-ip action."""
+        results = self._report_visits_by_ip()
+        event.set_results({"ips": results})
+
+    def _report_visits_by_ip(self) -> None:
+        """Report requests to nginx by IP and report action result."""
+        container = self.unit.get_container(CONTAINER_NAME)
+        log_file = container.pull("/var/log/nginx/access.log")
+        list_to_search = log_file.readlines()
+        log_file.close()
+        ip_regex = r"[0-9]+(?:\.[0-9]+){3}"
+        ip_list = []
+        for line in list_to_search:
+            line = line.split()
+            current_date = datetime.datetime.now()
+            date = datetime.datetime.strptime(line[3].lstrip("[").rstrip("]"), "%d/%b/%Y:%H:%M:%S")
```

Comment:
Can you rename this variable so that it reflects it's a `datetime` and not a `date`?